Data/BankerQueue/LazyLists.idr

```diff
@@ -0,0 +1,189 @@
+||| This is not actually a general lazy list library. It's really
+||| intended for implementing banker's queues. It also has some
+||| junk about regular lists for that purpose.
+module Data.BankerQueue.LazyLists
+
+%default total
+%access public
+
+||| A lazy-spined list.
+data LList : Type -> Type where
```

Comment:
CoList should be used when you want to signal to the totality checker that it should use productivity rather than reduction. I don't think that's the case here...